facebook/delphiFacebook/R/contingency_utils.R

```diff
@@ -0,0 +1,153 @@
+#' Update date and input files settings from params file
+#' 
+#' Use `end_date` and `aggregate_range` to calculate last full time period
+#' (either month or week) to use for aggregations. Find all files in `input_dir`
+#' within the calculated time period.
+#'
+#' @param params    Params object produced by read_params
+#'
+#' @return a named list of parameters values
+#'
+#' @importFrom lubridate ymd_hms as_date
+#' 
+#' @export
+update_params <- function(params) {
+  if ( is.null(params$end_date) & (is.null(params$input) | length(params$input) == 0) ) {
+    # Neither end_date nor list of input files is provided, assume want to use
+    # most current dates and data.
+    date_range <- get_range_prev_full_period(Sys.Date(), params$aggregate_range)
+    params$input <- get_filenames_in_range(date_range[[1]], date_range[[2]], params)
+  } else if ( !is.null(params$end_date) & (is.null(params$input) | length(params$input) == 0) ) {
+    # List of input files is not provided, assume want to use the full period
+    # preceding the provided end_date
+    date_range <- get_range_prev_full_period(
+      as_date(params$end_date), params$aggregate_range)
+    params$input <- get_filenames_in_range(date_range[[1]], date_range[[2]], params)
+  } else if ( is.null(params$end_date) & !is.null(params$input) & length(params$input) != 0 ) {
+    # Use list of input files provided, even if it does not constitute a full 
+    # period. Use dates in input files to select range.
+    date_range <- get_date_range_from_filenames(params)
+  } else if ( !is.null(params$end_date) & !is.null(params$input) & length(params$input) != 0 ) {
+    # Use provided date to determine range, and use provided list of input files.
+    date_range <- get_range_prev_full_period(
+      as_date(params$end_date), params$aggregate_range)
+  }
+
+  if (length(params$input) == 0) {
+    stop("no input files to read in")
+  }
+  
+  params$start_time <- date_range[[1]]
+  params$end_time <- date_range[[2]]
+
+  params$start_date <- as_date(date_range[[1]])
+  params$end_date <- as_date(date_range[[2]])
+
+  return(params)
+}
+
+#' Get relevant input data file names from `input_dir`.
+#'
+#' @param start_date    Start of desired date range 
+#' @param end_date    End of desired date range 
+#' @param params    Params object produced by read_params
+#'
+#' @return Character vector of filenames
+#' 
+#' @importFrom lubridate as_date days
+#' 
+#' @export
+get_filenames_in_range <- function(start_date, end_date, params) {
+  start_date <- as_date(start_date) - days(params$backfill_days)
+  end_date <- as_date(end_date)
+  date_pattern <- "^[0-9]{4}-[0-9]{2}-[0-9]{2}.*[.]csv$"
+  youtube_pattern <- ".*YouTube[.]csv$"
```

Comment:
The YouTube surveys are not used in any of the current pipelines; I don't think we need to support them here